datasets/winograd_wsc/winograd_wsc.py

```diff
@@ -0,0 +1,134 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Winograd Schema Challenge Dataset"""
+
+import xml.etree.ElementTree as ET
+
+import datasets
+
+
+_DESCRIPTION = """\
+A Winograd schema is a pair of sentences that differ in only one or two words and that contain an ambiguity that is
+resolved in opposite ways in the two sentences and requires the use of world knowledge and reasoning for its
+resolution. The schema takes its name from a well-known example by Terry Winograd:
+
+> The city councilmen refused the demonstrators a permit because they [feared/advocated] violence.
+
+If the word is ``feared'', then ``they'' presumably refers to the city council; if it is ``advocated'' then ``they''
+presumably refers to the demonstrators.
+"""
+
+_CITATION = """\
+@inproceedings{levesque2012winograd,
+  title={The winograd schema challenge},
+  author={Levesque, Hector and Davis, Ernest and Morgenstern, Leora},
+  booktitle={Thirteenth International Conference on the Principles of Knowledge Representation and Reasoning},
+  year={2012},
+  organization={Citeseer}
+}
+"""
+
+_HOMPAGE = "https://cs.nyu.edu/faculty/davise/papers/WinogradSchemas/WS.html"
+_DOWNLOAD_URL = "https://cs.nyu.edu/faculty/davise/papers/WinogradSchemas/WSCollection.xml"
+
+
+class WinogradWSCConfig(datasets.BuilderConfig):
+    """ BuilderConfig for WinogradWSC. """
+
+    def __init__(self, *args, language=None, inds=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.inds = set(inds) if inds is not None else None
+
+    def is_in_range(self, id):
+        """ Takes an index and tells you if it belongs to the configuration's subset """
+        return id in self.inds if self.inds is not None else True
+
+
+class WinogradWSC(datasets.GeneratorBasedBuilder):
+    """ The Winograd Schema Challenge Dataset """
+
+    BUILDER_CONFIG_CLASS = WinogradWSCConfig
+    BUILDER_CONFIGS = [
+        WinogradWSCConfig(
+            name="wsc285",
+            description="Full set of winograd examples",
+        ),
+        WinogradWSCConfig(
+            name="wsc273",
+            description="A commonly-used subset of examples. Identical to 'wsc285' but without the last 12 examples.",
+            inds=list(range(273)),
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "pronoun": datasets.Value("string"),
+                    "pronoun_loc": datasets.Value("int32"),
+                    "quote": datasets.Value("string"),
+                    "quote_loc": datasets.Value("int32"),
+                    "options": datasets.Sequence(datasets.Value("string")),
+                    "label": datasets.Value("int32"),
```

Comment:
Can you use the ClassLabel feature type here ?

Reply:
Sure yeah, as long as it doesn't matter that the meaning of the label changes from instance to instance (so having a label name doesn't make sense)